lib/rules/no-unassigned-vars.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * @fileoverview Rule to flag variables that are never assigned
+ * @author Jacob Bandes-Storch <https://github.com/jtbandes>
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/** @type {import('../types').Rule.RuleModule} */
+module.exports = {
+	meta: {
+		type: "problem",
+
+		defaultOptions: [
+			{
+				typeof: false,
+			},
+		],
+
+		docs: {
+			description:
+				"Disallow variable declarations that use `let` or `var` but have no intitial value and are never assigned",
+			recommended: true,
```

Comment:
```suggestion
			recommended: false,
```

Per our [semver policies](https://github.com/eslint/eslint?tab=readme-ov-file#semantic-versioning-policy), we can add rules to the recommended config only in major versions.

Reply:
Thanks, is there some procedure to recommend that this rule be considered for the next major version?

Reply:
Yes, before each major release we open an issue to discuss all rules that have been added and decide which ones will be added to the recommended config. Example: https://github.com/eslint/eslint/issues/17596.